lib/src/abi_specific_ints.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+
+// `char` in C.
+typedef Char = Int8;
+
+// `unsigned char` in C.
+typedef UnsignedChar = Uint8;
+
+// `short` in C.
+typedef Short = Int16;
+
+// `unsigned short` in C.
+typedef UnsignedShort = Uint16;
+
+// `int` in C.
+typedef Int = Int32;
```

Comment:
I think changing the reified type should be okay. (We can bump the major version.) I prefer having a typedef for now to not introduce unnecessary verbosity. But I don't feel very strong about it. If you prefer not using typedefs that's fine with me as well.

I've added tests in the SDK CL that check the size and signedness so that we notice if it diverges.